app/services/AccountService.js

```diff
@@ -98,4 +98,53 @@ exports.sendAccountRecoveryEmail = async (payload) => {
         data:response.Messages.ACCOUNTRECOVERY,
         statusCode:201
     }
+};
+
+exports.recoverPassword = async (payload) => {
+    const validator = accountValidator.validatePasswordRecoveryModel(payload);
+    if(validator){
+        return{
+            error: validator,
+            statusCode: 422
+        }
+    }
+
+    const token =  await tokenRepository.findOne({$and:[{token:payload.token},{email:payload.email}]});
+
+    if(!token) {
+        return {
+            error:response.Messages.ACCOUNTRECOVERYERROR,
+            statusCode:404
+        }
+    }
+    const expiryDate = new Date();
+
+    if(token._doc.createdAt < expiryDate.setMinutes(-5)) {
```

Comment:
Kindly extend the expiry date. 5 mins is too short. No be OTP. Put it at 20mins or so.